addons/mail/models/mail_thread.py

```diff
@@ -1409,6 +1409,9 @@ def _message_parse_extract_payload(self, message, save_original=False):
                     alternative = True
                 if part.get_content_type() == 'multipart/mixed':
                     mixed = True
+                if part.get_content_type() == 'bin/plain' and '.pdf' in part.get_filename():
+                    _logger.warning("Message containing an unexpected Content-Type, assuming 'pdf'")
+                    part.replace_header('Content-Type', part.get('ContentType', '').replace('bin/plain', 'application/pdf'))
```

Comment:
Globally would say this is the same as the code further up so maybe instead you could do
```py
original_content_type = part.get_content_type()
if original_content_type in ['binary/octet-stream', 'bin/plain']:
     new_content_type = 'application/octet-stream'
     if part.get_filename(failobj='').endswith('.pdf'):
          new_content_type = 'application/pdf'
     part.replace_header('Content-Type', new_content_type)
     _logger.warning("Message containing an unexpected Content-Type '%s', assuming '%s'", original_content_type, new_content_type)
```
So we can have one place where we "correct binary mimetypes"

As I think it's safe to assume bin/plain is at least octet-stream in all cases, and all .pdf files are likely actual pdfs